src/juno-node.ts

```diff
@@ -27,8 +30,30 @@ export default class JunoModule {
 		// this.connection.setOnDataListener(this.onDataHandler);
 	}
 
-	public static default(socketPath: string): JunoModule {
-		return new JunoModule(new SocketConnection(socketPath), new JsonProtocol());
+	public static async default(socketPath: string) {
+		const [ host, port ] = socketPath.split(':');
+
+		if (net.isIP(host) && typeof Number(port) === 'number') {
+			return this.fromInetSocket(host, Number(port));
+		}
+		if ( (await fs.promises.lstat(socketPath)).isSocket() ) {
```

Comment:
Although these checks are not being done in the libraries for other languages, I like the fact that you're doing these. But don't you think you can cover more edge cases if you do these checks in the specific functions of `fromInetSocket` and `fromUnixSocket`? So that even if those functions are called directly, these checks are still being executed.

Reply:
Yeah makes sense. I'll move the check for UnixSocket to `fromUnixSocket()`. But need to keep the _socketPath split and comparison_ in `default()` function, as it is a primary condition which separates InetSocket and UnixSocket calls.

Also, we won't need to worry about messed up parameter types as it'll be handled by ts. But for valid IP check, do I add the same condition in `fromUnixSocket()` function? Is there other cleaner solution?

Reply:
> Also, we won't need to worry about messed up parameter types as it'll be handled by ts. But for valid IP check, do I add the same condition in `fromUnixSocket()` function? Is there other cleaner solution?

We can all agree that JS is a very weird language. If you do `typeof Nan`, you'll get `undefined` as the answer, while if you do `typeof NaN`, you'll get `number` as the answer. So that check as such won't work. You'll have to do `isNan(Number(port))`. Fucking JS.

As for your question, it doesn't make sense to check of IPs in `fromUnixSocket` since that's expecting a socket file. But yes, since we're doing checks anyway, I'd recommend doing it all the way and not stop half-way. So I'd recommend doing the `isIP` check inside `fromInetSocket` as well.

Reply:
Ouch. JS Sneak 100.

Okayy then, I'll add the checks as mentioned